src/table/__tests__/TableBodyWrapper.spec.jsx

```diff
@@ -0,0 +1,65 @@
+import './rtl-setup';
+import React from 'react';
+import { render } from '@testing-library/react';
+import { expect } from 'chai';
+import sinon from 'sinon';
+
+import { generateDataPages, generateLayout } from './generate-test-data';
+import manageData from '../handle-data';
+
+import TableBodyWrapper from '../TableBodyWrapper';
+import * as selectionsUtils from '../selections-utils';
+import * as renderer from '../cells/renderer';
+
+describe('<TableBodyWrapper />', async () => {
+  const sandbox = sinon.createSandbox();
+  const model = { getHyperCubeData: async () => generateDataPages(2, 2) };
+
+  let tableData;
+  let constraints;
+  let selectionsAPI;
+
+  beforeEach(async () => {
+    sandbox.replace(selectionsUtils, 'addSelectionListeners', () => {});
+
+    tableData = await manageData(model, generateLayout(1, 1), { top: 0, height: 100 });
+    constraints = {};
+    selectionsAPI = {};
+  });
+
+  afterEach(() => {
+    sandbox.verifyAndRestore();
+    sandbox.resetHistory();
+  });
+
+  it('should render 2x2 table body without CellRenderer', () => {
+    sandbox.replace(renderer, 'getCellRenderer', () => {});
+
+    const { queryByText } = render(
+      <TableBodyWrapper tableData={tableData} constraints={constraints} selectionsAPI={selectionsAPI} />
+    );
+
+    expect(queryByText(tableData.rows[0]['id-0'].qText)).to.be.visible;
+    expect(queryByText(tableData.rows[0]['id-1'].qText)).to.be.visible;
+    expect(queryByText(tableData.rows[1]['id-0'].qText)).to.be.visible;
+    expect(queryByText(tableData.rows[1]['id-1'].qText)).to.be.visible;
+  });
+  it('should render 2x2 table body and call CellRenderer', () => {
```

Comment:
I'm a bit confused here on what you exactly want to test between these two.

Reply:
If you look at the logic in `TableBodyWrapper`, there are two cases for rendering cells. If there is no CellRenderer for that column, it uses the mui TableCell, hence two tests. Both test should render a table body, but using different components

Reply:
Hmm now I see that the sandbox replace is different indeed. However from getCellRenderer you will always get back something right? Default seems to be the TableCell as well.

Reply:
Hmm yeah, getCellRenderer used to only return something for dimensions, but now it always returns something. So might be able to remove the logic all together. I'll keep it for now, but will have it in mind when we add column styling etc